src/compiler/checker.ts

```diff
@@ -37014,7 +37014,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             if (!witnesses) {
                 return false;
             }
-            const operandConstraint = getBaseConstraintOrType(checkExpressionCached((node.expression as TypeOfExpression).expression));
+            const operandConstraint = getBaseConstraintOfType(checkExpressionCached((node.expression as TypeOfExpression).expression)) || unknownType;

```

Comment:
Note that the constraint was already obtained here before so this doesn't really change that much. It just unifies how `T` and `T extends unknown` are handled here (`switch (typeof x)` was already treated as exhaustive for the latter).

On top of that, this is literally what `getTypeFactsWorker` (that is used later here) does as the very first thing. So this change doesn't even affect the other branch of the code in a meaningful way.

Reply:
I had to stare at this diff for a good 10 seconds before I even realized it was calling a different function. 😵‍💫  They differ by only a single character (and `r` and `f` look quite similar at small font sizes) and it feels like it would be really easy to mix these up when reading the code.